### ISSUE TITLE
ELECTRON BAD Versions List

### ISSUE THREAD

**Issue by theofficialgman** [Botspot/pi-apps#1930]
keep your eyes on these upstream projects as they are known to use electron:
tabby
turbowarp
whatsapp
electron fiddle
webcord
armcord

gmans electron blacklist:
electron 22.0.0+ (bad wayland)
electron 19.0.0-19.0.4 (too high glibc)
electron 18.0.0-18.2.3 (too high glibc)
electron < 13.5.0 (bad on bullseye/jammy and newer glibc)

![image](https://user-images.githubusercontent.com/28281419/170603180-b4b4a65a-3fdd-4221-9ab5-9f2f493965f7.png)

**Comment by ryanfortner** [comment]
Another day, another Electron bug. :)

**Comment by Rak1ta** [comment]
3.2.0 installed on Buster and started
[install-success-Discord.log](https://github.com/Botspot/pi-apps/files/8810526/install-success-Discord.log)

**Comment by theofficialgman** [comment]
> 3.2.0 installed on Buster and started [install-success-Discord.log](https://github.com/Botspot/pi-apps/files/8810526/install-success-Discord.log)

it shouldn't start unless you have a franken debian and have glibc 2.29+, which you should not have on buster (glibc 2.28)
```
ldd --version
```

and 
```
echo '// Part of iq.OpenGenus.org
#include <stdio.h>
#include <stdlib.h>
#include <gnu/libc-version.h>

int main(int argc, char *argv[])
{
    // Print glibc version
    printf("GNU libc version: %s\n", gnu_get_libc_version());
    exit(EXIT_SUCCESS);
}
' > glibc.cpp
g++ -std=c++11 glibc.cpp -o glibc
./glibc
```

**Comment by theofficialgman** [comment]
also it is quite possible that only ARM64 libffmpeg by chance depended on libc 2.29, and that armhf libffmpeg did not. regardless... this is still an issue and upstream is building on focal which caused it

**Comment by theofficialgman** [comment]
> also it is quite possible that only ARM64 libffmpeg by chance depended on libc 2.29, and that armhf libffmpeg did not. regardless... this is still an issue and upstream is building on focal which caused it

yes this is the case. due to compiler chance x86_64 and armhf only depend on libc 2.17+ while arm64 depends on libc 2.29+

**Comment by theofficialgman** [comment]
https://github.com/electron/electron/pull/34491

Progress ^

**Comment by theofficialgman** [comment]
merged into electron 19.x.y and 20.x.y, will be fixed in their next releases

**Comment by theofficialgman** [comment]
merged into electron 19.x.y and 20.x.y, will be fixed in their next releases

**Comment by theofficialgman** [comment]
electron 19.0.5 is being built and will be available shortly https://github.com/electron/electron/commit/0764bb1560411d663f747427eca410e309b2b040

**Comment by theofficialgman** [comment]
Scratch that, 19.0.6 now because 19.0.5 timed out 😆

**Comment by Sussy-OS** [comment]
Any updates on this yet?

**Comment by theofficialgman** [comment]
nope not yet... both 19.0.6 and 19.0.5 timed out and the release numbers were reverted. this is probably pushed back to the end of this week.
just watch (click the watch button) at electron to get notified when a new release is made https://github.com/electron/electron

**Comment by theofficialgman** [comment]
electron 19.0.5 has released a few hours ago

@smartfrigde @SpacingBat3 @cycool29 please rebuild your respective applications on electron 19.0.5+ as 19.0.0-19.0.4 are broken releases on arm64 linux buster/bionic (glibc 2.29+ was required)

**Comment by theofficialgman** [comment]
closing as the affected projects have been bumped to 19.0.5

**Comment by smartfrigde** [comment]
ArmCord 3.1.2 has reverted back to Electron 21, if possible please update to it.
